pkg/vault/vault.go

```diff
@@ -92,14 +93,15 @@ func NewInternalVault(name string, serviceAccountName string, namespace string)
 	data[ServiceAccount] = serviceAccountName
 	data[Namespace] = namespace
 
-	return FromMap(data, namespace)
+	vault, _, err := FromMap(data, namespace)
+	return vault, err
 }
 
 // FromMap reads the configuration of a Vault instance from a map.
 // defaultNamespace is used when there is no namespace value provided in the map (for backwards compatibility reasons).
-func FromMap(data map[string]string, defaultNamespace string) (Vault, error) {
+func FromMap(data map[string]string, defaultNamespace string) (Vault, bool, error) {
```

Comment:
On first sight I am not a big fan of functions which return multiple concerns. Why so you need `FromMap` to return this extra boolean, if you can call `FromMap` and then `IsExternal`? This is a very slippery slope and leads imo to very intertangled APIs.

Reply:
that was an oversight, will remove that

Reply:
> @cagiti could you create an issue for it, explaining where and how the problem manifests itself? I am not quite sure atm what we are trying to fix here.

yes, does every PR need a issue?

Reply:
> yes, does every PR need an issue?

that's what we said, yes.

Reply:
the most important thing is really that there is enough context to understand the context of the issue. It just helps tremendously when reviewing the code. 

Then there is a debate between providing context on the pr or an issue. Given that the latter for example drives the changelogs (at least it should), every change should be backed by an issue. 

For me, there is another reason for it. Even though pull request and issues seem similar on GitHub they are not. Assume you create a pull request and you nicely add the stack trace which cause your trouble into the pull request which fixes the issue. If someone else get the same error and decides to search whether there this is a known issue, he would initially go to the issues page and paste the search string there. Per default, you are only searching issues though so the fact that this error occurred already once and might be fixed in a later version of `jx` is lost.

Reply:
Seems there was an existing issue: https://github.com/jenkins-x/jx/issues/7315